### PR TITLE
mgokv: add Update method

### PR DESCRIPTION
**Pull request by rogpeppe**
This allows atomic updates to a value.

### CODE REVIEW

File: mgokv/persist.go [hunk 1]
```diff
@@ -121,6 +121,31 @@ func (s *Session) PutInitial(key string, val interface{}) (bool, error) {
...
+//
+// will atomically increment the N value.
+//
+// This function returns the error with the same cause
+// that mgo.Collection.Update returns.
+func (s *Session) Update(key string, update interface{}) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if err := s.coll.UpdateId(key, update); err != nil {
+		return errgo.Mask(err, errgo.Any)
```
Comment (mhilton) [inline mgokv/persist.go]: should this not return ErrNotFound as opposed to mgo.ErrNotFound?

Comment (mhilton) [inline mgokv/persist.go]: you'll also want to test this case

Comment (rogpeppe) [inline mgokv/persist.go]: Good idea. Done.

**Comment by rogpeppe** [comment]
$$merge$$

**Pull request merged**